experimental/traffic-portal/src/app/core/statuses/status-details/status-details.component.ts

```diff
@@ -0,0 +1,175 @@
+
+import { ServerService } from "src/app/api";
+import { DecisionDialogComponent, DecisionDialogData } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+/**
+ * StatusDetailsComponent is the controller for a status "details" page.
+ */
+@Component({
+	providers: [ServerService],
+	selector: "tp-status-details",
+	styleUrls: ["./status-details.component.scss"],
+	templateUrl: "./status-details.component.html",
+})
+export class StatusDetailsComponent implements OnInit {
+
+	/** Status ID expected from the route param using which we identify whether we are creating new status or load existing status */
+	public id: string | number | null = null;
+
+	/** All details of status requested */
+	public statusDetails: ResponseStatus | null = null;
+
+	/** Reactive form intialized to creat / edit status details */
+	public statusDetailsForm!: FormGroup;
+
+	/** Loader status for the actions */
+	public loading = false;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param serverService The Servers API which is used to provide row data.
+	 * @param route A reference to the route of this view which is used to get the 'id' query parameter of status.
+	 * @param router Angular router
+	 * @param dialog Dialog manager
+	 * @param fb Form builder
+	 * @param navSvc Manages the header
+	 */
+	constructor(
+		private readonly serverService: ServerService,
+		private readonly route: ActivatedRoute,
+		private readonly router: Router,
+		private readonly dialog: MatDialog,
+		private readonly navSvc: NavigationService,
+	) { }
+
+	/** Initializes table data, loading it from Traffic Ops. */
+	public ngOnInit(): void {
+		this.statusDetailsForm = new FormGroup({
+			description: new FormControl("",Validators.required),
+			name: new FormControl("",Validators.required),
+		});
+
+		// Getting id from the route
+		this.id = this.route.snapshot.paramMap.get("id");
+
+		// we check whether params is a number if not we shall assume user wants to add a new status.
+		if (!this.isNew) {
+			this.loading = true;
+			this.statusDetailsForm.addControl("id", new FormControl(""));
+			this.statusDetailsForm.addControl("lastUpdated", new FormControl(""));
+			this.getStatusDetails();
+		} else {
+			this.navSvc.headerTitle.next("New Status");
+		}
+	}
+
+	/**
+	 * Reloads the servers table data.
+	 *
+	 * @param id is the id passed in route for this page if this is a edit view.
+	 */
+	public async getStatusDetails(): Promise<void> {
+		const id = Number(this.id) ; // id Type 'null' is not assignable to type 'string'
+		this.statusDetails = await this.serverService.getStatuses(id);
+		const data: ResponseStatus = {
+			description: this.statusDetails.description,
+			id: this.statusDetails.id,
+			lastUpdated: new Date(),
+			name: this.statusDetails.name
+		};
+
+		// Set page title with status ID
+		this.navSvc.headerTitle.next(`Status #${data.id}`);
```

Comment:
The title should probably be the name of the status, instead of or in addition to its ID. Nobody knows what status #4 is, but everyone knows what `ONLINE` is.